core/mapping/utils.go

```diff
@@ -311,6 +311,10 @@ func parseNumberRange(str string) (*numberRange, error) {
 		right = math.MaxFloat64
 	}
 
+	if left >= right {
```

Comment:
Good but not enough, I think.

For example
- `[2:2]` is valid
- `[2:2)` is invalid
- `(2:2]` is invalid
- `(2:2)` is invalid

Reply:
ok, changed [2:2]  to valid